src/software/field_tests/field_test_fixture.py

```diff
@@ -0,0 +1,412 @@
+                AssignedTacticPlayControlParams, tactics
+            )
+        else:
+            self.yellow_full_system_proto_unix_io.send_proto(
+                AssignedTacticPlayControlParams, tactics
+            )
+
+    def set_play(self, play: Play, isBlue):
+        if team == proto.ssl_gc_common_pb2.Team.BLUE:
+            self.blue_full_system_proto_unix_io.send_proto(Play, play)
+
+        else:
+            self.yellow_full_system_proto_unix_io.send_proto(Play, play)
+
+    def send_gamecontroller_command(
+        self,
+        gc_command: proto.ssl_gc_state_pb2.Command,
+        team: proto.ssl_gc_common_pb2.Team,
+        final_ball_placement_point=None,
+    ):
+
+        self.gamecontroller.send_ci_input(
+            gc_command=gc_command,
+            team=team,
+            final_ball_placement_point=final_ball_placement_point,
+        )
+
+    def set_worldState(self, worldstate: WorldState):
+        def __validate(eventually_validation_set, timeout):
+            timeout_time = time.time() + timeout
+
+            while time.time() < timeout_time:
+                try:
+                    current_world = self.world_buffer.get(
+                        block=True, timeout=WORLD_BUFFER_TIMEOUT
+                    )
+
+                    (
+                        eventually_validation_status,
+                        always_validation_status,
+                    ) = validation.run_validation_sequence_sets(
+                        world=current_world,
+                        eventually_validation_sequence_set=[
+                            [eventually_validation_set]
+                        ],
+                        always_validation_sequence_set=[[]],
+                    )
+
+                    if not validation.contains_failure(eventually_validation_status):
+                        break
+
+                except queue.Empty as empty:
+                    logger.warning("failed to obtain world")
+
+            validation.check_validation(eventually_validation_status)
+
+        BLUE_TEAM_INDEX = 0
+        YELLOW_TEAM_INDEX = 1
+
+        logger.info("testing ids matching")
+
+        # validate that ids match with test setup
+        ids_present = True
+        for idx, robot_ids in enumerate(
+            [worldstate.blue_robots, worldstate.yellow_robots]
+        ):
+            team_name = "friendly" if idx == 0 else "enemy"
+            team_robots = (
+                self.friendly_robot_ids_field
+                if idx == 0
+                else self.enemy_robot_ids_field
+            )
+            for robot_id in robot_ids:
+                if robot_id not in team_robots:
+                    logger.warning(
+                        f"robot {id} from the {team_name} team present in test but not on the field"
+                    )
+                    ids_present = False
+
+        if not ids_present:
+            raise Exception("robotIds do not match")
```

Comment:
Ignoring this would mean the test wont function as expected

Reply:
what would be the dangers of waiving this exception? I'm thinking about the situation where we may have a few robots being repaired, so we only have 5/6 robots available and it'd be nice to still run the field test

Reply:
There is no danger per se but waiving the exception means that whatever robots the test wants to work with aren't on the field. The result is that we will be sending commands to no one for the ids that arent on the field. To me it seems like a logical thing to have as I can imagine someone being completely puzzled as to why the robots arent moving. It is also there to prevent false positives (ex you think the test passed, but it actually only controlled one robot when it was supposed to use 2 robots). 

I understand the situation you proposed though, and its the same kind of thinking which motivated the second ticket that could be opened mentioned above: "allow robot id mapping from test to on field id, this would save us time from having to manually change test parameters at the time of testing."

Perhaps a good workaround for now is to have a cmd line argument to bypass this check? this way you only proceed if you know what you are doing.

Reply:
the cmd line arg alternative to bypass this makes sense to me for testing